README.md

```diff
@@ -387,6 +387,90 @@ if (!string.IsNullOrWhiteSpace(claimsChallenge))
 
 ```
 
+### Using Session State
+
+When a ChallengeUser method is called, the redirect URL is pointing to GET method by default. When the challenge happen during Create POST, for example, user will have a bad experience after being redirected to Create GET. One of the ways to smoothen user experience is to use session state to store fields information.
+Take a look into the example of using session state.
```

Comment:
When the `ChallengeUser()` method is called, the redirect URL of the request sent to Azure AD will bring the user back to the /GET path. So the user might get confused if the authentication context was applied to an action like /CREATE. To make for a smoother user experience, we use the [ASP.NET Session state](https://docs.microsoft.com/aspnet/core/fundamentals/app-state) to store the user's input. This allows us to restore the ToDo item once the user has come back to the web app after redirection.